File: 2025/07 July/23 Sum of Subarrays/main.cpp
// Link: https://www.geeksforgeeks.org/problems/sum-of-subarrays2229/1

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int subarraySum(vector<int> &arr)
    {
        int n = arr.size();
        int result = 0;

        for (int i = 0; i < n; i++)
            result += (arr[i] * (i + 1) * (n - i));

        return result;
    }
};

int main()
{
    Solution sol;
    vector<int> arr = {1, 2, 3};
    int result = sol.subarraySum(arr);
    cout << "Sum of all subarrays: " << result << endl;

    arr = {1, 3};
    result = sol.subarraySum(arr);
    cout << "Sum of all subarrays: " << result << endl;

    return 0;
}

File: 2025/07 July/23 Sum of Subarrays/main.py
# Link: https://www.geeksforgeeks.org/problems/sum-of-subarrays2229/1


class Solution:
    def subarraySum(self, arr):
        n = len(arr)
        result = 0

        for i in range(n):
            result += arr[i] * (i + 1) * (n - i)

        return result


if __name__ == "__main__":
    sol = Solution()
    arr = [1, 2, 3]
    result = sol.subarraySum(arr)
    print(f"Sum of all subarrays: {result}")

    arr = [1, 3]
    result = sol.subarraySum(arr)
    print(f"Sum of all subarrays: {result}")
